librapidsvn/src/folder_browser.cpp

```diff
@@ -182,6 +182,11 @@ struct FolderBrowser::Data
   ~Data()
   {
     DeleteAllItems();
+	  if (singleContext != 0)
+	  {
+		  delete singleContext;
+		  singleContext = 0;
+	  }
   }
```

Comment:
This can by accomplished by:
```c++
delete singleContext;
singleContext = 0;
```

Reply:
I think `if` statement is needed.
when `singleContext == 0`, `delete singleContext` can have problem.
Am I missing something?

Reply:
It's perfectly fine to use `delete` on a `nullptr` (see https://en.cppreference.com/w/cpp/language/delete).
But the `singleContext = 0` is needed so a second call does not try to delete the pointer again.

Reply:
if the case is the same, the same action is valid (haven't checked with the code).

Reply:
See:
https://isocpp.org/wiki/faq/freestore-mgmt#delete-handles-null
https://isocpp.org/wiki/faq/freestore-mgmt#double-delete-disaster